### DISCUSSION TITLE
Generating beatmaps from any song

### DISCUSSION THREAD

**Discussion by ghost** [ppy/osu#19969] [Q&A] [answered]
I've always been wondering if there have ever been any discussions or plans to have a feature where you can basically generate a beatmap from any custom song that you upload as an MP3 etc.?

I'm not suggesting for this to be added to osu!; I'm just wondering because I imagine that in theory it should be possible to analyze a song, detect the beats, and then on every beat perhaps place a circle, or maybe detect two beats close to each other and put a slider starting at the first beat and ending at the second beat? Things like that.

As for the detection of a beat, I think you could get the frequency data at the current point of the song, get the average frequency, compare it to the last average and then compare the difference to some threshold and then you could decide whether that is a beat or not. Not sure about that.

And as for the way the circles and sliders are laid out, perhaps it could pick from a few predefined samples and mix those together randomly to make up a beatmap.

Or maybe in the far far future AI could be used to do this.

In practice I'm sure this is extremely complicated to get right. I'm just curious.

**Comment by peppy** [comment, marked as answer]
Not in scope. Half of the fun in osu! is creating the beatmap. You're welcome to attempt this in a separate project.

**Reply by ghost** [reply]
Sure, that makes sense!
